src/linear_algebra/unitary_group.lean

```diff
@@ -140,6 +142,60 @@ to `general_linear_group n α`. -/
 def embedding_GL : unitary_group n α →* general_linear_group α (n → α) :=
 ⟨λ A, to_GL A, by simp, by simp⟩
 
+section norm
+variables {𝕜 : Type*} [is_R_or_C 𝕜]
+local attribute [instance] matrix.normed_group
+
+open_locale big_operators
+
+/-- The entrywise sup norm of a unitary matrix is at most 1. -/
+lemma entrywise_sup_norm_bound_of_unitary {U : matrix n n 𝕜} (hU : U ∈ matrix.unitary_group n 𝕜) :
+  ∥ U ∥ ≤ 1 :=
+begin
+  rw pi_norm_le_iff zero_le_one,
+  intro i,
+  rw pi_norm_le_iff zero_le_one,
+  intro j,
+  -- The norm squared of an entry is at most the L2 norm of its row.
+  have norm_sum : ∥ U i j ∥^2 ≤ (∑ (x : n), ∥ U i x ∥^2),
+  { apply multiset.single_le_sum,
+  { intros x h_x,
+    rw multiset.mem_map at h_x,
+    cases h_x with a h_a,
+    rw ← h_a.2,
+    apply sq_nonneg },
+  { rw multiset.mem_map,
+    use j,
+    simp only [eq_self_iff_true, finset.mem_univ_val, and_self, sq_eq_sq] } },
+  -- The L2 norm of a row is a diagonal entry of U ⬝ Uᴴ
+  have diag_eq_norm_sum : (U ⬝ Uᴴ) i i = ∑ (x : n), ∥ U i x ∥^2,
+  { simp only [matrix.mul_apply, matrix.conj_transpose_apply, ← star_ring_end_apply],
+    conv_lhs
+    begin
+      congr,
+      skip,
+      funext,
+      rw [is_R_or_C.mul_conj, is_R_or_C.norm_sq_eq_def'],
+    end,
+    norm_cast },
```

Comment:
```suggestion
  { simp only [matrix.mul_apply, matrix.conj_transpose_apply, ←star_ring_end_apply,
               is_R_or_C.mul_conj, is_R_or_C.norm_sq_eq_def', is_R_or_C.of_real_pow] },
```
Unlike `rw`, `simp` can rewrite inside a bunch of binders, so in a situation like this it's always worth a try.